src/transformStateWithClones.js

```diff
@@ -7,7 +7,23 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformHistory = [];
+  let transformStep = { ...state };
+
+  for (const element of actions) {
+    if (element.type === 'addProperties') {
+      Object.assign(transformStep, element.extraData);
+    } else if (element.type === 'removeProperties') {
+      for (const removeKeys of element.keysToRemove) {
+        delete transformStep[removeKeys];
+      }
+    } else if (element.type === 'clear') {
+      transformStep = {};
+    }
+    transformHistory.push({ ...transformStep });
```

Comment:
If I'm not wrong, you can push the object directly, no need for destrucuring

```suggestion
    transformHistory.push(transformStep);
```

Reply:
Yeah you can push it directly, but what will happen if we change some values in this object afterwards?)

Reply:
@illiarerun nothing good 💯 we already had conversation with Dima about that 👍